firebase-sessions/src/test/kotlin/com/google/firebase/sessions/SessionInitiatorTest.kt

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.sessions
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+
+class SessionInitiatorTest {
+  class FakeTime {
+    var currentTimeMs = 0L
+      private set
+
+    fun addTimeMs(intervalMs: Long) {
+      currentTimeMs += intervalMs
+    }
+  }
+
+  class SessionStartCounter {
+    var count = 0
+      private set
```

Comment:
Is this a way to denote `count` to have a setter?

Reply:
Yeah this is some Kotlin syntax sugar. var means it has a getter and setter of the same visibility by default, public here. Then I overrode the setter to be private.

Reply:
Thanks for the clear explanation. Seems to a very useful construct. Should we use this construct in other places of our code?